src/main/java/com/google/sps/classes/Utility.java

```diff
@@ -41,17 +41,19 @@
   // Query to retrieve data from all questions. Can be appended a WHERE condition to select
   // specific questions. Generates the following table:
   //
-  // |-----------------Question-----------------|----FollowerCount--------|-----GetUsername-----|------AnswerCount------|
-  // +----+-------+------+----------+-----------+-------------+-----------+----------+----------+-------------+---------+
-  // | id | title | body | asker_id | date_time | question_id | followers | username | asker_id | question_id | answers |
-  // +----+-------+------+----------+-----------+-------------+-----------+----------+----------+-------------+---------+
+  // |-----------------Question-----------------|----FollowerCount--------|-----GetUsername-----|------AnswerCount------|----UserFollows---|
+  // +----+-------+------+----------+-----------+-------------+-----------+----------+----------+-------------+---------+------------------+
+  // | id | title | body | asker_id | date_time | question_id | followers | username | asker_id | question_id | answers | follows_question |
+  // +----+-------+------+----------+-----------+-------------+-----------+----------+----------+-------------+---------+------------------+
   public static final String fetchQuestionsQuery = "SELECT * FROM Question "
       + "LEFT JOIN (SELECT question_id, COUNT(follower_id) followers FROM QuestionFollower "
       + "GROUP BY question_id) FollowerCount ON Question.id=FollowerCount.question_id "
       + "LEFT JOIN (SELECT username, id AS asker_id FROM User) GetUsername "
       + "ON Question.asker_id=GetUsername.asker_id "
       + "LEFT JOIN (SELECT question_id, COUNT(id) answers FROM Answer "
-      + "GROUP BY question_id) AnswerCount ON Question.id=AnswerCount.question_id ";
+      + "GROUP BY question_id) AnswerCount ON Question.id=AnswerCount.question_id "
+      + "LEFT JOIN (SELECT question_id AS follows_question FROM QuestionFollower WHERE follower_id=?) "
+      + "UserFollows ON Question.id=UserFollows.follows_question ";
```

Comment:
I don't really understand the follows_question column. From its name, I suppose its a boolean that determines if the logged in user follows the question, but from the query, this column seems to be an id of the question?

Reply:
Ok, from what you added in lines 171 and 172 I understand now, but this might be very confusing, so I think it needs clarification, so please add comment either here or below.

Reply:
alright, I added a comment on line 172, is it ok?